microsoft/testsuites/kdump/kdumpcrash.py

```diff
@@ -0,0 +1,263 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from pathlib import PurePosixPath
+from typing import cast
+
+from assertpy import assert_that
+
+from lisa import (
+    LisaException,
+    Logger,
+    Node,
+    RemoteNode,
+    SkippedException,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    UnsupportedDistroException,
+    node_requirement,
+    schema,
+    search_space,
+    simple_requirement,
+)
+from lisa.operating_system import Redhat
+from lisa.tools import Dmesg, Echo, KdumpBase, Uname
+from lisa.util.shell import wait_tcp_port_ready
+
+
+@TestSuiteMetadata(
+    area="kdump",
+    category="functional",
+    description="""
+    This test suite is used to check if kernel crash dmup is effect, which is judged by
+    if vmcore file is generated after triggering kdump by sysrq.
+
+    Steps:
+    1. Check if vmbus version, kernel configurations is supported for crash dump.
+    2. Specify the memory reserved for kdump kernel in kernel cmdline, setting the
+         crashkernel= option to the required value.
+        a. Modify the grub config file to add crashkrenel option or change the value to
+            the required one. (For Redhat 8, no need to modify grub config file. It can
+            specify crashkernel by using grubby command directly)
+        b. Update grub config
+    3. Reboot system to make kdump effect.
+    4. Check if the crash kernel is loaded.
+        a. Check if kernel cmdline has crashkernel option and the value is expected.
+        b. Check if /sys/kernel/kexec_crash_loaded file exists and the value is '1'.
+        c. Check if /proc/iomem is reserved memory for crash kernel
+    5. Trigger kdump through 'echo c > /proc/sysrq-trigger'
+    6. Check if vmcore or dump file is generated under'/var/crash' after system boot up.
+
+    It has 7 test cases in this test suite. They verfy if the kdump is effect when:
+        1. with different vcpu count, 1, or more than 2, 16.
+        2. the crashkernel is set auto.
+        3. with a large count cpu, such as VM size E96a_v4.
+        4. with a large memory, such as M128s, whose memory is 2T.
+    """,
+)
+class KdumpCrash(TestSuite):
+    crash_kernel = "512M"
+    timeout_of_dump_crash = 60
+
+    @TestCaseMetadata(
+        description="""
+        This test case verfiy if the kdump is effect when VM has one vcpu.
+        The VM need 2G memory to make sure there are enough memory for crash kernel
```

Comment:
What happens, if the memory is not enough?

Reply:
The crash kernel may not be loaded.

Reply:
Ok, please specify it's "for load the crash kernel", it's clearer than "for crash kernel"

Reply:
Done